sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -142,3 +246,141 @@ Azure::Response<DeletedKey> KeyClient::GetDeletedKey(
+          return _detail::DeletedKeySinglePageDeserialize(rawResponse);
+        },
+        {_detail::DeletedKeysPath});
+  }
+  // Get next page //
+  Azure::Core::Url nextPageUrl(options.ContinuationToken.GetValue());
+  auto query = nextPageUrl.GetQueryParameters();
+  if (options.MaxResults)
+  {
+    query.emplace("maxResults", std::to_string(options.MaxResults.GetValue()));
+  }
+  return m_pipeline->SendRequest<DeletedKeySinglePage>(
+      context,
+      Azure::Core::Http::HttpMethod::Get,
+      [](Azure::Core::Http::RawResponse const& rawResponse) {
+        return _detail::DeletedKeySinglePageDeserialize(rawResponse);
+      },
+      {nextPageUrl.GetPath()},
+      query);
+}
+
+Azure::Response<PurgedKey> KeyClient::PurgeDeletedKey(
+    std::string const& name,
+    Azure::Core::Context const& context) const
+{
+  return m_pipeline->SendRequest<PurgedKey>(
+      context,
+      Azure::Core::Http::HttpMethod::Delete,
+      [](Azure::Core::Http::RawResponse const&) { return PurgedKey(); },
+      {_detail::DeletedKeysPath, name});
+}
+
+Azure::Response<KeyVaultKey> KeyClient::UpdateKeyProperties(
+    KeyProperties const& properties,
+    Azure::Nullable<std::list<KeyOperation>> const& keyOperations,
+    Azure::Core::Context const& context) const
+{
+  return m_pipeline->SendRequest<KeyVaultKey>(
+      context,
+      Azure::Core::Http::HttpMethod::Patch,
+      _detail::KeyRequestParameters(properties, keyOperations),
+      [&properties](Azure::Core::Http::RawResponse const& rawResponse) {
+        return _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(properties.Name, rawResponse);
+      },
+      {_detail::KeysPath, properties.Name, properties.Version});
+}
+
+Azure::Response<std::vector<uint8_t>> KeyClient::BackupKey(
+    std::string const& name,
+    Azure::Core::Context const& context) const
+{
+  // Use the internal model KeyBackup to parse from Json
+  auto response = m_pipeline->SendRequest<_detail::KeyBackup>(
+      context,
+      Azure::Core::Http::HttpMethod::Post,
+      [](Azure::Core::Http::RawResponse const& rawResponse) {
+        return _detail::KeyBackup::Deserialize(rawResponse);
+      },
+      {_detail::KeysPath, name, "/backup"});
+
+  // Convert the internal KeyBackup model to a raw vector<uint8_t>.
+  return Azure::Response<std::vector<uint8_t>>(
+      response.ExtractValue().Value, response.ExtractRawResponse());
+}
+
+Azure::Response<KeyVaultKey> KeyClient::RestoreKeyBackup(
+    std::vector<uint8_t> const& backup,
+    Azure::Core::Context const& context) const
+{
+  _detail::KeyBackup backupModel;
+  backupModel.Value = backup;
+  return m_pipeline->SendRequest<KeyVaultKey>(
+      context,
+      Azure::Core::Http::HttpMethod::Post,
+      backupModel,
+      [](Azure::Core::Http::RawResponse const& rawResponse) {
+        return _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(rawResponse);
+      },
+      {_detail::KeysPath, "restore"});
```

Comment:
Why is restore -> "restore" but backup is "/backup"?  One of them wrong :)

Reply:
/backup is wrong!
It works because calling microsoft.com/something and microsoft.com//something gets resolved to the same thing.